### PR TITLE
MCR 3174 fix pmd issues in mycore sword

### PR DESCRIPTION
**Pull request by Antoniafriedrich**
[Link to jira](https://mycore.atlassian.net/browse/MCR-3174).

### CODE REVIEW

File: mycore-sword/src/main/java/org/mycore/sword/MCRSwordUtil.java [hunk 1]
```diff
@@ -572,10 +576,6 @@ static void addEditMediaLinks(String collection, DepositReceipt depositReceipt,
         }
     }
 
-    public interface MCRFileValidator {
```
Comment (toKrause) [inline mycore-sword/src/main/java/org/mycore/sword/MCRSwordUtil.java:575]: Why has this public interface  been removed while fixing `AvoidReassigningParameters`?

File: mycore-sword/src/main/java/org/mycore/sword/MCRSwordUtil.java [hunk 2]
```diff
@@ -178,7 +176,6 @@ public static MediaResource getZippedDerivateMediaResource(String object) {
     }
 
     private static void addDirectoryToZip(ZipArchiveOutputStream zipOutputStream, Path directory) {
-        MCRSession currentSession = MCRSessionMgr.getCurrentSession();
```
Comment (toKrause) [inline mycore-sword/src/main/java/org/mycore/sword/MCRSwordUtil.java:181]: `MCRSessionMgr#getCurrentSession()` has a potential side effect. Not sure if removing the method call is okay here.

File: mycore-sword/src/main/java/org/mycore/sword/MCRSwordUtil.java (hunk 1 above)
Comment (Antoniafriedrich) [reply mycore-sword/src/main/java/org/mycore/sword/MCRSwordUtil.java:575]: Just checked it. It seems to be in line 376 now. I am not sure why it is marked as deleted here but it is still there.

File: mycore-sword/src/main/java/org/mycore/sword/MCRSwordUtil.java (hunk 2 above)
Comment (Antoniafriedrich) [reply mycore-sword/src/main/java/org/mycore/sword/MCRSwordUtil.java:181]: The variable was targeted by UnusedLocalVariable PMD rule as well as greyed out by IntelliJ but if it has negative side effects I can leave it in and surpress the warning. Would that be better?

Comment (toKrause) [reply mycore-sword/src/main/java/org/mycore/sword/MCRSwordUtil.java:181]: Or just call `MCRSessionMgr.getCurrentSession();` and ignore the return value.

Not sure if there are side effect, but the call could be intentional. That's why I just commented and didn't request a change. Someone with more knowledge should make that decision.

**Review by sebhofmann — APPROVED** [review]

**Pull request merged**